testutils/resourcebuilder/machine/v1beta1/aws_provider_spec.go

```diff
@@ -20,48 +20,90 @@ import (
 	"encoding/json"
 
 	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	"github.com/openshift/cluster-api-actuator-pkg/testutils/resourcebuilder"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/ptr"
 )
 
-// AWSProviderSpec creates a new AWS machine config builder.
-func AWSProviderSpec() AWSProviderSpecBuilder {
-	return AWSProviderSpecBuilder{
-		availabilityZone: "us-east-1a",
-		instanceType:     "m6i.xlarge",
-		securityGroups: []machinev1beta1.AWSResourceReference{
-			{
-				Filters: []machinev1beta1.Filter{
-					{
-						Name: "tag:Name",
-						Values: []string{
-							"aws-security-group-12345678",
-						},
-					},
-				},
-			},
-		},
-		subnet: machinev1beta1.AWSResourceReference{
+var (
+	defaultInstanceType          = "m6i.xlarge"
+	defaultAvailabilityZone      = "us-east-1a"
+	defaultRegion                = "us-east-1"
+	defaultUserDataSecretName    = "aws-user-data-12345678"
+	defaultCredentialsSecretName = "aws-cloud-credentials"
+	defaultAMIID                 = ptr.To[string]("aws-ami-12345678")
```

Comment:
Surprised this is a pointer, does it need to be, for the coalesce to work correctly?